gcloud/test_connection.py

```diff
@@ -374,7 +395,11 @@ def __init__(self, authorized=None):
 
     def authorize(self, http):
         self._called_with = http
-        return self._authorized
+
+        if callable(self._authorized):
```

Comment:
because the way the mock was done it would return the same object from `authorize` no matter how many times you called or which threads, so I wasn't able to test in the thread whether it was returning a unique object.